src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/AssemblyPart.cs

```diff
@@ -41,6 +51,25 @@ public AssemblyPart(Assembly assembly)
         /// <inheritdoc />
         public IEnumerable<TypeInfo> Types => Assembly.DefinedTypes;
 
+        public PrecompiledViews PrecompiledViews
+        {
+            get
+            {
+                var precompiledAssemblyName = new AssemblyName(Assembly.FullName);
+                precompiledAssemblyName.Name = precompiledAssemblyName.Name + PrecompiledViewsAssemblySuffix;
+
+                var viewFactoryTypeName = $"{ViewFactoryNamespace}.{ViewFactoryTypeName},{precompiledAssemblyName}";
```

Comment:
Sure. I was initially was playing around with trying to codegen the type factory we create in https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCacheResult.cs#L47 which is why I came up with this name. We could still do that here.